sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -130,18 +170,35 @@ class SQLContext(@transient val sparkContext: SparkContext)
   protected[sql] lazy val optimizer: Optimizer = DefaultOptimizer
 
   @transient
-  protected[sql] val ddlParser = new DDLParser(sqlParser.apply(_))
-
-  @transient
-  protected[sql] val sqlParser = {
-    val fallback = new catalyst.SqlParser
-    new SparkSQLParser(fallback(_))
+  protected[sql] val ddlParser = new DDLParser((sql: String) => { getSQLDialect().parse(sql) })
+  protected[sql] def dialectCache = tlSession.get().dialectCache
```

Comment:
I know we talked about this briefly, but its still unclear to me why you need this.  You are still making reflective calls on a per statement basis?  It seems unlikely to me that caching the result of `newInstance` is really that important.  Can you clarify and if its absolutely not possible to remove this explain why in a comment.